s2a_options.go

```diff
@@ -125,6 +145,12 @@ type ClientOptions struct {
 
 	// Optional fallback after dialing with S2A fails.
 	FallbackOpts *FallbackOptions
+
+	// Generates S2AStream for talking to the S2A server.
+	GetS2AStream *func() (*S2AStream, error)
```

Comment:
Any reason we need to return `*func()...` instead of `func()...`? IIUC, the latter is more common.

Reply:
Changed to `func()…`.